sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -1218,15 +1219,32 @@ def merge(
     merged = frame_base.DeferredFrame.wrap(
         expressions.ComputedExpression(
             'merge',
-            lambda left, right: left.merge(
-                right, left_index=True, right_index=True, **kwargs),
+            lambda left, right: left.merge(right,
+                                           left_index=True,
+                                           right_index=True,
+                                           suffixes=suffixes,
+                                           **kwargs),
             [indexed_left._expr, indexed_right._expr],
             preserves_partition_by=partitionings.Singleton(),
             requires_partition_by=partitionings.Index()))
 
     if left_index or right_index:
       return merged
     else:
+      common_cols = set(left_on).intersection(right_on)
+      if len(common_cols):
+        # When merging on the same column name from both dfs, merged will have
+        # two duplicate columns, one with lsuffix and one with rsuffix.
+        # Normally pandas de-dupes these into a single column with no suffix.
+        # This replicates that logic by dropping the _right_ dupe, and removing
+        # the suffix from the _left_ dupe.
+        lsuffix, rsuffix = suffixes
+        merged = merged.drop(
```

Comment:
What if `{col}{rsuffix}` or `{col}{lsuffix}` already exists in the input?

Reply:
In that case I think it would have been renamed to {col}{rsuffix}{rsuffix} - this is a good edge case to think about though. I'll look at adding some more test cases.

Reply:
Ah nevermind you're right, pandas keeps {col}{rsuffix}. hmm

Reply:
Thanks for catching this. I updated the logic to drop the duplicate column(s) from the right df proactively